server/src/controllers/attainable.ts

```diff
@@ -0,0 +1,120 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import { Request, Response } from 'express';
+import * as yup from 'yup';
+
+import models from '../database/models';
+import Attainable from '../database/models/attainable';
+import Course from '../database/models/course';
+import CourseInstance from '../database/models/courseInstance';
+
+import { ApiError } from '../types/error';
+import { idSchema } from '../types/general';
+import { HttpCode } from '../types/httpCode';
+import { findAttainableById } from './utils/attainable';
+import { findCourseById } from './utils/course';
+import { findCourseInstanceById } from './utils/courseInstance';
+
+const requestSchema: yup.AnyObjectSchema = yup.object().shape({
```

Comment:
I was going to suggest defining this in the controller below, but this schema is actually used by multiple controllers. In that case, since this is global it should be renamed to be more explicit like `addEditAttainableRequestSchema` (bad name, just an example).